datastream-server-api/src/main/java/com/linkedin/datastream/server/DatastreamProducerRecord.java

```diff
@@ -117,9 +124,22 @@ public Optional<Integer> getPartition() {
     return _partition;
   }
 
+  /**
+   * Set partition for the record. Can be used to set partition for broadcast record each time before sending.
+   * @param partition Topic partition to send the record to.
+   */
+  public void setPartition(int partition) {
```

Comment:
I think we need to keep this class immutable. Are you just trying to avoid a copy? I think that's not worth the broken semantics.